components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3700,4 +3700,20 @@ public static String getJwkThumbPrint(Certificate certificate) throws IdentityOA
             throw new IdentityOAuth2Exception("Error occurred while generating SHA-1 JWK thumbprint", e);
         }
     }
+
+    /**
+     * Delete all access tokens of a given tenant id.
+     *
+     * @param tenantId Id of the tenant
+     * @throws IdentityOAuth2Exception
+     */
+    public static void deleteAllAccessTokens(int tenantId) throws IdentityOAuth2Exception {
+
+        OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO().deleteAccessTokensByTenantId(tenantId);
+
+        OAuthCache.getInstance().clear();
```

Comment:
This would clear the cache of the whole server right?

Reply:
Yes. That will happen since the cache is started in a super tenant flow.  Any alternative ways to clean tenant specific tokens?

Reply:
I think that logic there inside revokeTokens(tenantId) ?? @sumedhe can you check?

Reply:
Sure 👍